### ISSUE TITLE
CA1822 doesn't suggest method be made static if called from xUnit fact

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn-analyzers#3019]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers)

#### Package Version

v2.9.7

#### Diagnostic ID

CA1822 Mark members as static

#### Repro steps

```c#
using Xunit;

public class Program
{
    [Fact]
    private void M()
    {
        N();
    }

    private void N() // expected suggestion to convert to static
    {
    }
}
```

#### Expected behavior

Suggests method `N` should be static.

#### Actual behavior

No suggestion.

Removing the `FactAttribute` from `M` causes the suggestion to appear as expected.

**Comment by Evangelink** [comment]
Hi @drewnoakes,

I am most likely missing something from your example because I don't see why the suggestion of having `N` marked as `static` would cause any issue.

**Comment by mavasani** [comment]
@Evangelink I believe the repro suggests that CA1822 is expected to be generated on method N, but is actually not reported. We are likely bailing out completely on types that have any test method.

**Comment by Evangelink** [comment]
The problem is that we only report for `private` and `internal` members when they were invoked via a method call. But as we exclude the analysis of some methods we might miss some invocations.
